shard_consumer.go

```diff
@@ -250,4 +251,31 @@ mainloop:
 		}
 		iterator = next
 	}
+	// Handle checkpointer updates which occur after a stop request comes in (whose originating records were before)
```

Comment:
Would be good to get a follow up ticket in here to go for a formal shutdown hook as opposed to a time based last ditch solution.

Reply:
Noting this here for reference when I get round to addressing feedback, as I'm focused on other things right now. I've thought of an approach that's way simpler than what I tried before:

At the same time that we get the last event timestamp (when we grab the record from the stream), we should also record `latestSequenceNumber`.

Then, when we get a stop request and break the main loops, we check if the `cp.sequenceNumber == latestSequenceNumber`.

If it does, we've got all the cp.updates that we need. So commit and return.

If it doesn't, proceed into the loop, but for every ticker perform that check, and once it evaluates to true commit and exit.

Keep the timeout of max age/2, to avoid hanging if the user app is just borked.